monitoring/bin/deploy_monitoring_tenant.sh

```diff
@@ -0,0 +1,170 @@
+kubectl delete secret --ignore-not-found -n $VIYA_NS prometheus-federate-$VIYA_TENANT
+kubectl create secret generic \
+  -n $VIYA_NS \
+  prometheus-federate-$VIYA_TENANT \
+  --from-file cluster-federate-job=$tenantDir/mt-federate-secret.yaml
+
+# Deploy Prometheus
+log_info "Deploying Prometheus"
+kubectl apply -n $VIYA_NS -f $tenantDir/mt-prometheus.yaml
+
+# Grafana password
+if helm3ReleaseExists grafana-$VIYA_TENANT $VIYA_NS; then
+  log_info "Upgrading Grafana via Helm...($(date) - timeout 10m)"
+else
+  grafanaPwd="$GRAFANA_ADMIN_PASSWORD"
+  if [ "$grafanaPwd" == "" ]; then
+    log_debug "Generating random Grafana admin password..."
+    showPass="true"
+    grafanaPwd="$(randomPassword)"
+  fi
+  log_info "Deploying Grafana via Helm...($(date) - timeout 10m)"
+fi
+
+# Deploy Grafana
+GRAFANA_CHART_VERSION_TENANT=${GRAFANA_CHART_VERSION_TENANT:-6.9.1}
+helm upgrade --install $helmDebug \
+  -n "$VIYA_NS" \
+  -f "$wnpGrafanaValuesFile" \
+  -f "$grafanaYAML" \
+  -f "$grafanaTLSYAML" \
+  -f "$userGrafanaYAML" \
+  --set "extraLabels.sas\\.com/tenant=$VIYA_TENANT" \
+  --set grafana.adminPassword="$grafanaPwd" \
+  --version "$GRAFANA_CHART_VERSION_TENANT" \
+  --atomic \
+  --timeout "10m" \
+  $extraArgs \
+  grafana-$VIYA_TENANT \
+  grafana/grafana
+
+# Deploy ServiceMonitors
+kubectl apply -n $VIYA_NS -f $tenantDir/serviceMonitor-sas-cas-tenant.yaml
+kubectl apply -n $VIYA_NS -f $tenantDir/serviceMonitor-sas-pushgateway-tenant.yaml
+
+function deploy_tenant_dashboards {
+   dir=$1
+   
+   log_message "--------------------------------"
+   for f in $dir/*.json; do
+     # Need to check existence because if there are no matching files,
+     # f will include the wildcard character (*)
+     if [ -f "$f" ]; then
+       log_debug "Deploying dashboard from file [$f]"
+       name=$(basename $f .json)
+      
+       kubectl create cm -n $DASH_NS $name --dry-run=client --from-file $f -o yaml | kubectl apply -f -
+       kubectl label cm -n $DASH_NS $name --overwrite grafana_tenant_dashboard=1 sas.com/monitoring-base=kube-viya-monitoring
+     fi
+   done
+   log_message "--------------------------------"
+}
+
+# Deploy dashboards
+DASH_NS=$VIYA_NS
+DASH_BASE=$tenantDir/dashboards
+deploy_tenant_dashboards monitoring/multitenant/dashboards
+
+if [ "$showPass" == "true" ]; then
+  # Find the grafana pod
+  grafanaPod="$(kubectl get po -n $VIYA_NS -l app.kubernetes.io/name=grafana --template='{{range .items}}{{.metadata.name}}{{end}}')"
+
+  log_notice "====================================================================="
+  log_notice "Generated Grafana admin password is: $grafanaPwd"
+  log_notice "Change the password at any time by running (replace password):       "
+  log_notice "kubectl exec -n $VIYA_NS $grafanaPod -c grafana -- bin/grafana-cli admin reset-admin-password myNewPassword"
+  log_notice "====================================================================="
+  log_message ""
+fi
```

Comment:
Doesn't look like we show the URL for the tenant-specific Grafana like we do when running deploy_monitoring_cluster.sh; should we?